app/src/store/store.ts

```diff
@@ -1,212 +1,165 @@
+                    dispatch('postsModule/setSelectedPost', post)
+                      .then(() =>
+                        // to be sure that the param types are not overriden in the query
+                        dispatch("updateURIFromState"));
+                  }
+                }));
+        });
     },
+
     /**
      * Updates url parameter with currently values from the store
      */
     updateURIFromState({ state }): void {
       const query = {
         ...router.currentRoute.query,
-        q: state.textSearchModule.searchValues.join(","),
-        area: state.international ? "international" : "national",
-        location: state.locationSearchModule.selectedLocation
-          ? state.locationSearchModule.selectedLocation.title
-          : "",
-        radius: state.locationSearchModule.selectedRadius,
-        page: state.page.toString(),
+        q: state.searchModule.searchValues.join(","),
+        area: state.searchModule.international ? "international" : "national",
+        location: state.searchModule.selectedLocation ? state.searchModule.selectedLocation.title : "",
+        radius: state.searchModule.selectedRadius,
+        page: state.postsModule.selectedPage.toString(),
       };
 
       let path = "/posts";
       // is there a post currently open? => reflect it in the route
-      if (state.selectedPost) {
-        path += "/" + state.selectedPost.id;
-      }
+      if (state.postsModule.selectedPost)
+        path += "/" + state.postsModule.selectedPost.id;
 
-      router
-        .push({
+      // only change route if query parameter change from current query parameter
+      if (JSON.stringify(query) !== JSON.stringify(router.currentRoute.query) || router.currentRoute.path !== path)
+        router.push({
           path,
-          query,
+          query
+        });
+    },
+
+    /**
+     *  find posts from DataService by setted parameter 
+     */
+    loadPosts({ state, commit }): Promise<Post[]> {
+      return DataService.findBySelection({
+        searchValues: state.searchModule.searchValues,
+        location: state.searchModule.selectedLocation,
+        radius: state.searchModule.selectedRadius,
+        from: state.postsModule.resultsFrom,
+        size: state.postsModule.resultSetSize,
+        international: state.searchModule.international,
+      })
+        .then((result: PaginatedResponse<Post>) => {
+          state.postsModule.totalResultSize = result.meta.total;
+          commit("postsModule/setPosts", result.data);
+          return result.data;
         })
-        .catch((err) => err);
+        .then((posts: Post[]) => {
+          // there is a full list of posts
+          if (posts.length) {
+            state.radiusExtended = state.radiusExtendedFrom ? true : false;
+
+            if (state.radiusExtendedFrom !== state.searchModule.selectedRadius
+              && !state.searchModule.selectedRadius)
+              state.radiusExtended = false;
+
+            return posts;
```

Comment:
See other comments about async await

Reply:
I split the functionality in blocks to make them more understandable.

If you have such problems with promises, i could do a presentation about promises.

Reply:
Im fully aware of how promises work.
My main point of using async/await is cleaner code. It's more readable and preferable over nested .thens().
https://hackernoon.com/6-reasons-why-javascripts-async-await-blows-promises-away-tutorial-c7ec10518dd9 sums up the benefits of async/await quite well.